proxy/proxy.go

```diff
@@ -673,7 +687,7 @@ func WithParams(p Params) *Proxy {
 
 	return &Proxy{
 		routing:                  p.Routing,
-		roundTripper:             tr,
+		roundTripper:             p.CustomHttpRoundTripperWrap(tr),
```

Comment:
looks good to me, just started thinking whether we should make more dynamic by calling the wrapper on every case where the roundtripper is used, and not only on initialization. Not sure yet what would be the implications. Also, we can do that in a subsequent changeset.

Reply:
which cases do you mean by "every case"?

Reply:
by every case I meant every outgoing request, but currently this is just a thought.